model/textparse/protobufparse.go

```diff
@@ -171,7 +161,7 @@ func (p *ProtobufParser) Series() ([]byte, *int64, float64) {
 	// away from gogo-protobuf to an actively maintained protobuf
 	// implementation. Once that's done, we can simply use the `optional`
 	// keyword and check for the unset state explicitly.
-	return p.metricBytes.Bytes(), nil, v
+	return p.seriesBytes.Bytes(), nil, v
```

Comment:
Speculation, for future investigation: can we use `dec.mData` here?

Reply:
Yes, I want to unpack this later. I think we can just you might miss cache when switching protocols in the single scrape loop (is this possible? e.g.  same target negotiating different things). This might leak memory then.

We might need some handling for this case, otherwise this would allow us to NOT parse labels on every scrape for cache hits.

Reply:
Playing with this idea, but the blocker might be the fact that [proto serialization is not canonical](https://protobuf.dev/programming-guides/serialization-not-canonical/#:~:text=Unfortunately%2C%20protobuf%20serialization%20is%20not,of%20proto%20serialization%20as%20unstable.)

Reply:
Some options:

a) We only need "canonical" stability per scrape target. We can build guards e.g. another map from labels hash to cache key to remove duplicate entries. This will handle case of switching content type in the same target life time.
b) We add to proto hash (identity) and ask clients to give us this
c) We spend CPU to parse proto first for hash/cache key, then parse that to labels only if needed.

Reply:
I think the only problem with missing the cache is the stats for metrics get incremented. Same thing happens if you reorder labels in text format (there is an issue about this).